src/components/engagement/engagement.service.ts

```diff
@@ -15,54 +26,823 @@ import {
+      canCreate: true,
+    };
   }
 
   async createLanguageEngagement(
-    _input: CreateLanguageEngagement,
-    _session: ISession
+    { languageId, projectId, ...input }: CreateLanguageEngagement,
+    session: ISession
   ): Promise<LanguageEngagement> {
-    throw new NotImplementedException();
+    try {
+      // Initial LanguageEngagement
+      const id = generate();
+      const acls = {
+        canReadLanguage: true,
+        canEditLanguage: true,
+        canReadProject: true,
+        canEditProject: true,
+        canReadProduct: true,
+        canEditProduct: true,
+        canReadFirstScripture: true,
+        canEditFirstScripture: true,
+        canReadLukePartnership: true,
+        canEditLukePartnership: true,
+        canReadSentPrintingDate: true,
+        canEditSentPrintingDate: true,
+        canReadStatus: true,
+        canEditStatus: true,
+        canReadCeremony: true,
+        canEditCeremony: true,
+        canReadCompleteDate: true,
+        canEditCompleteDate: true,
+        canReadDisbursementCompleteDate: true,
+        canEditDisbursementCompleteDate: true,
+        canReadCommunicationsCompleteDate: true,
+        canEditCommunicationsCompleteDate: true,
+        canReadStartDate: true,
+        canEditStartDate: true,
+        canReadEndDate: true,
+        canEditEndDate: true,
+        canReadInitialEndDate: true,
+        canReadLastSuspendedAt: true,
+        canReadLastReactivatedAt: true,
+        canReadStatusModifiedAt: true,
+        canReadModifiedAt: true,
+      };
+
+      const ceremony = await this.ceremonyService.create(
+        { type: CeremonyType.Dedication },
+        session
+      );
+
+      await this.db.createNode({
+        type: LanguageEngagement.classType,
+        session,
+        input: {
+          id,
+          ...input,
+        },
+        acls,
+        aclEditProp: 'canCreateEngagement',
+      });
+
+      // connect Language and Project to LanguageEngagement.
+      const query = `
+        MATCH
+          (project:Project {id: $projectId, active: true}),
+          (language:Language {id: $languageId, active: true}),
+          (ceremony:Ceremony {id: $ceremonyId, active: true}),
+          (languageEngagement:LanguageEngagement {id: $id, active: true})
+        CREATE
+          (project)-[:engagement {active:true, createAt: datetime()}]->(languageEngagement),
+          (languageEngagement)-[:language {active: true, createAt: datetime()}]->(language),
+          (languageEngagement)-[:ceremony {active: true, createAt: datetime()}]->(ceremony)
+        RETURN languageEngagement.id as id
+      `;
+      await this.db
+        .query()
+        .raw(query, {
+          languageId: languageId,
+          projectId: projectId,
+          ceremonyId: ceremony.id,
+          id: id,
+        })
+        .first();
+
+      const res = (await this.readOne(id, session)) as LanguageEngagement;
+      return res;
+    } catch (e) {
+      this.logger.error(e);
```

Comment:
Unfortunately types cannot be specified in catch statements, so `e` is always `any`.
Nonetheless, errors currently cannot be passed directly to log statements. 

```suggestion
      this.logger.error('Failed to create language engagement', { exception: e });
```

Same for the rest of the calls in this file.